include/picongpu/plugins/shadowgraphy/Shadowgraphy.hpp

```diff
@@ -0,0 +1,633 @@
+                    {
+                        ys[i] = helper->getY(i);
+                    }
+
+                    ::openPMD::MeshRecordComponent yMRC = spatialMesh["y"];
+                    yMRC.setPosition(std::vector<double>{0.0});
+
+                    ::openPMD::Datatype datatype_y = ::openPMD::determineDatatype<float_X>();
+                    ::openPMD::Extent extent_y = {static_cast<unsigned long int>(helper->getSizeY()), 1};
+                    ::openPMD::Dataset dataset_y = ::openPMD::Dataset(datatype_y, extent_y);
+                    yMRC.resetDataset(dataset_y);
+
+                    // Write actual data
+                    ::openPMD::Offset offset_y = {0};
+                    yMRC.storeChunk(ys, offset_y, extent_y);
+
+
+                    series.iterations[currentStep].close();
+                }
+
+                //! Write Fourier output to openPMD file
+                void writeFourierOutputToOpenPMDFile(uint32_t currentStep)
+                {
+                    std::stringstream filename;
+                    filename << m_help->optionFileName.get(m_id) << "_fourierdata_%T."
+                             << m_help->optionFileExtention.get(m_id);
+                    ::openPMD::Series series(filename.str(), ::openPMD::Access::CREATE);
+
+                    auto meshNeg = series.iterations[currentStep].meshes["Fourier Domain Fields - negative"];
+                    meshNeg.setGeometry(::openPMD::Mesh::Geometry::cartesian);
+                    meshNeg.setDataOrder(::openPMD::Mesh::DataOrder::C);
+                    meshNeg.setGridSpacing(std::vector<double>{1.0, 1.0, 1.0});
+                    meshNeg.setGridGlobalOffset(
+                        std::vector<double>{static_cast<double>(helper->getOmegaIndex(0)), 0.0, 0.0});
+                    meshNeg.setGridUnitSI(1.0);
+                    meshNeg.setAxisLabels(std::vector<std::string>{
+                        "Spatial x index",
+                        "Spatial y index",
+                        "Fourier transform frequency index"});
+                    meshNeg.setUnitDimension(std::map<::openPMD::UnitDimension, double>{
+                        {::openPMD::UnitDimension::L, 1.0},
+                        {::openPMD::UnitDimension::M, 1.0},
+                        {::openPMD::UnitDimension::T, -3.0},
+                        {::openPMD::UnitDimension::I, -1.0}});
+
+                    // Reshape abstract MeshRecordComponent
+                    ::openPMD::Datatype datatype = ::openPMD::determineDatatype<std::complex<float_64>>();
+                    ::openPMD::Extent extent
+                        = {static_cast<unsigned long int>(helper->getNumOmegas() / 2),
+                           static_cast<unsigned long int>(helper->getSizeY()),
+                           static_cast<unsigned long int>(helper->getSizeX())};
+                    ::openPMD::Offset offset = {0, 0, 0};
+
+                    // Go through all 8 different fields components
```

Comment:
I do not understand that comment. What are the 8 components?

Reply:
these https://github.com/ComputationalRadiationPhysics/picongpu/pull/4868/files#diff-6f072cdd2bf72a84ca9a6f25d1239e2d382d66c92a59351c2e09dbf871ef58cfR511 ?

Reply:
The link for you does not for for me. But it's +Ex, +Ey, +Bx, +By and the negative ones

Reply:
Added a comment

Reply:
Thanks